src/OpenCensus/Implementation/OpenCensusEventSource.cs

```diff
@@ -41,6 +41,27 @@ public void ExporterThrownExceptionWarning(string ex)
             this.WriteEvent(1, ex);
         }
 
+        [Event(2, Message = "{0} should be an ASCII string with a length greater than 0 and not exceeding {1} characters.", Level = EventLevel.Warning)]
+        public void InvalidCharactersInElement(string element)
+        {
+            this.WriteEvent(2, element, Constants.MaxResourceTypeNameLength);
+        }
+
+        [NonEvent]
+        public void SecurityExceptionWarning(Exception ex)
+        {
+            if (Log.IsEnabled(EventLevel.Warning, EventKeywords.All))
+            {
+                this.SecurityExceptionWarning(ToInvariantString(ex));
+            }
+        }
+
+        [Event(3, Message = "Main library failed with security exception: {0}")]
```

Comment:
It is a good practice to have specialized messages, not generic ones. This will narrow down what's wrong for customer

Reply:
I'll refine